serverless/src/libs/bunnyCDN.ts

```diff
@@ -0,0 +1,261 @@
+      data: {
+        Hostname: options.hostname,
+      },
+    });
+
+    return true;
+  }
+
+  public async removeCustomHostname(options: AddCustomHostnameMethodArgs) {
+    await this.fetchBunny(`/pullzone/${options.pullZoneId}/removeHostname`, {
+      method: 'DELETE',
+      data: {
+        Hostname: options.hostname,
+      },
+    });
+
+    return true;
+  }
+
+  public async loadFreeCertificate(options: LoadFreeCertificateMethodArgs) {
+    await this.fetchBunny(
+      `/pullzone/loadFreeCertificate?hostname=${options.hostname}`,
+      {
+        method: 'GET',
+      }
+    );
+
+    return true;
+  }
+
+  public async setForceSSL(options: SetForceSSLMethodArgs) {
+    await this.fetchBunny(`/pullzone/${options.pullZoneId}/setForceSSL`, {
+      data: {
+        Hostname: options.hostname,
+        ForceSSL: options.shouldForceSSL ?? true,
+      },
+    });
+
+    return true;
+  }
+
+  public async purgePullZoneCache(options: PurgePullZoneCacheMethodArgs) {
+    await this.fetchBunny(`/pullzone/${options.pullZoneId}/purgeCache`, {});
+
+    return true;
+  }
+}
+
+export type ErrorData = {
+  ErrorKey: string;
+  Field: string;
+  Message: string;
+};
+
+export type GetPullZoneMethodArgs = {
+  pullZoneId: string;
+};
+
+export type CreatePullZoneMethodArgs = {
+  zoneId: string;
+  originUrl: string;
+};
+
+export type UpdatePullZoneMethodArgs = {
+  pullZoneId: string;
+  originUrl: string;
+};
+
+export type DeletePullZoneMethodArgs = {
+  pullZoneId: string;
+};
+
+export type AddCustomHostnameMethodArgs = {
+  pullZoneId: string;
+  hostname: string;
+};
+
+export type LoadFreeCertificateMethodArgs = {
+  hostname: string;
+};
+
+export type RemoveCustomHostnameMethodArgs = {
+  pullZoneId: string;
+  hostname: string;
+};
+
+export type SetForceSSLMethodArgs = {
+  pullZoneId: string;
+  hostname: string;
+  shouldForceSSL?: boolean;
+};
+
+export type PurgePullZoneCacheMethodArgs = {
+  pullZoneId: string;
+};
+
+type HostnameInterface = {
+  /** The unique ID of the hostname */
+  Id: number;
+  /** The hostname value for the domain name */
+  Value: string;
+  /** Determines if the Force SSL feature is enabled */
+  ForceSSL: string;
+  /** Determines if this is a system hostname controlled by bunny.net */
+  IsSystemHostname: boolean;
+  /** Determines if the hostname has an SSL certificate configured */
+  HasCertificate: true;
+};
+
+type PullZoneData = {
+  Id: number;
+  Name: string;
+  OriginUrl: string;
+  Hostnames: HostnameInterface[];
+};
+
+type FetchPullZoneWhenNameAlreadyTakenArgs = {
+  name: string;
+};
+
+export const fetchPullZoneWhenNameAlreadyTaken =
+  ({ name }: FetchPullZoneWhenNameAlreadyTakenArgs) =>
+  async (error: BunnyCdnError | AxiosError<ErrorData>) => {
+    if (
+      error instanceof BunnyCdnError &&
+      error.name === 'pullzone.name_taken'
+    ) {
+      return fetchPullZone({ name });
+    }
+
+    throw error;
```

Comment:
this code seems to be repeated with the previous comment, a possible enhancement would be a single function that repeats for this given error.

Reply:
this function returns the pullzone but in the previous comment's code snippet we continue to the next loop iteration.
Actually, this function should be removed since we don't use it anywhere.